SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -204,25 +170,65 @@ private void runLiDAR()
             {
                while(startRead)
                {
-            	   if(amountOfBytes() >= 9)
-            	   {
-            		   byte bytes[] = readBytes();
-            		   if(isValid(bytes))
-            		   {
-            			   distL = bytes[2];
-            			   distH = bytes[3];
-            			   strengthL = bytes[4];
-            			   strengthH = bytes[5];
-            			   reservedBytes = bytes[6];
-            			   signalQualityDegree = bytes[7];
-            		   }
-            	   }
+                   byte bytes[] = sensor.read(1);
+                   if(bytes.length > 0)
+                   {
+                       switch(state)
+                       {
+                           case WAIT_START1:
+                               if(bytes[0] == 0x59)
+                               {
+                                   count = 0;
+                                   checksum = bytes[0];
+                                   packet[0] = bytes[0];
+                                   state = ReceivingState.WAIT_START2;
+                               }
+                               break;
+                           case WAIT_START2:
+                               if(bytes[0] == 0x59)
+                               {
+                                   count++;
+                                   checksum += bytes[0];
+                                   packet[1] = bytes[0];
+                                   state = ReceivingState.RECEIVING;
+                               }
+                               else
+                               {
+                                   state = ReceivingState.WAIT_START1;
+                               }
+                               break;
+                           case RECEIVING:
+                               count++;
+                               packet[count] = bytes[0];
+                               if(count == PACKET_LENGTH - 1)
+                               {
+                                   // Reached the end of the packet, check checksum
+                                   if(checksum == bytes[0])
+                                   {
+                                       parsePacket();
```

Comment:
For now, we may want to calculate the checksum but not prevent packet parse if it is bad, just to see if we are receiving some data.

Reply:
We save every packet in the packet[] member regardless of whether it's good or bad. This could be public and dumped to the dashboard, though, for debug purposes.